pkg/security/ebpf/probes/event_types.go

```diff
@@ -181,34 +189,38 @@ var SelectorsPerEventType = map[eval.EventType][]manager.ProbesSelector{
 			&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "kprobe/vfs_mkdir"}},
 			&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "kprobe/filename_create"}},
 		}},
-		&manager.AllOf{Selectors: ExpandSyscallProbesSelector(
-			manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "mkdir"}, EntryAndExit),
-		},
-		&manager.AllOf{Selectors: ExpandSyscallProbesSelector(
-			manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "mkdirat"}, EntryAndExit),
-		},
+		&manager.OneOf{Selectors: []manager.ProbesSelector{
+			&manager.OneOf{Selectors: ExpandSyscallProbesSelector(
+				manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "mkdir"}, EntryAndExit),
+			},
+			&manager.OneOf{Selectors: ExpandSyscallProbesSelector(
+				manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "mkdirat"}, EntryAndExit),
+			},
+		}},
 	},
 
 	// List of probes to activate to capture open events
 	"open": {
 		&manager.AllOf{Selectors: []manager.ProbesSelector{
 			&manager.ProbeSelector{ProbeIdentificationPair: manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "kprobe/vfs_truncate"}},
 		}},
-		&manager.AllOf{Selectors: ExpandSyscallProbesSelector(
-			manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "open"}, EntryAndExit, true),
-		},
-		&manager.AllOf{Selectors: ExpandSyscallProbesSelector(
-			manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "creat"}, EntryAndExit),
-		},
-		&manager.AllOf{Selectors: ExpandSyscallProbesSelector(
-			manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "open_by_handle_at"}, EntryAndExit, true),
-		},
-		&manager.AllOf{Selectors: ExpandSyscallProbesSelector(
-			manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "truncate"}, EntryAndExit, true),
-		},
-		&manager.AllOf{Selectors: ExpandSyscallProbesSelector(
-			manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "openat"}, EntryAndExit, true),
-		},
+		&manager.OneOf{Selectors: []manager.ProbesSelector{
+			&manager.OneOf{Selectors: ExpandSyscallProbesSelector(
+				manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "open"}, EntryAndExit, true),
+			},
+			&manager.OneOf{Selectors: ExpandSyscallProbesSelector(
+				manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "creat"}, EntryAndExit),
+			},
+			&manager.OneOf{Selectors: ExpandSyscallProbesSelector(
+				manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "open_by_handle_at"}, EntryAndExit, true),
+			},
+			&manager.OneOf{Selectors: ExpandSyscallProbesSelector(
+				manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "truncate"}, EntryAndExit, true),
+			},
+			&manager.OneOf{Selectors: ExpandSyscallProbesSelector(
+				manager.ProbeIdentificationPair{UID: SecurityAgentUID, Section: "openat"}, EntryAndExit, true),
+			},
+		}},
```

Comment:
I initially thought it was a nit, but maybe not: I think we should group `OneOf` only if we're sure that the `OneOf`s in question might be false. Like I understand why you would group `openat` and `openat` since afaik `openat2` was recently added to the kernel, but I don't think we should group `open` with `creat` for example. If we can't hook onto `creat` but we can hook on `open` we won't be notified ...

Reply:
Same applies to all the other syscalls unfortunately ...